proxywasm/internal/abi_callback_configuration_test.go

```diff
@@ -57,16 +58,23 @@ func Test_proxyOnVMStart(t *testing.T) {
 		contextIDToRootID: map[uint32]uint32{},
 	}
 
+	// Check ABI version. There is no return value so just make sure it doesn't panic.
+	proxyABIVersion()
+
 	// Call OnVMStart.
 	proxyOnVMStart(0, 0)
 	require.True(t, vmContext.onVMStartCalled)
 	require.Equal(t, uint32(0), currentState.activeContextID)
 
-	// Check ABI version. There is no return value so just make sure it doesn't panic.
-	proxyABIVersion()
-
-	// Allocate memory
-	require.NotNil(t, proxyOnMemoryAllocate(100))
+	// Allocate memory. The returned buffer has the same size as passed in
+	bufPtr := proxyOnMemoryAllocate(100)
+	buf := unsafe.Slice(bufPtr, 101)
+	for _, b := range buf[:100] {
+		require.Equal(t, byte(0), b)
+	}
+	require.Panics(t, func() {
+		_ = buf[101]
```

Comment:
shouldn't this be 100?
```suggestion
		_ = buf[100]
```